MapboxNavigation/RouteVoiceController.swift

```diff
@@ -182,16 +187,16 @@ open class RouteVoiceController: NSObject, AVSpeechSynthesizerDelegate, AVAudioP
         return true
     }
     
-    func speak(_ text: String, error: String? = nil) {
+    func speak(_ text: String, error: Error?) {
         // Note why it failed
         if let error = error {
-            print(error)
+            voiceControllerDelegate?.spokenInstructionsDidFail?(self, error: error)
```

Comment:
We need to reapply the refactor that was proposed in #624: the caller should be responsible for handling errors. Making this method the bottleneck for error reporting is a premature optimization.

Reply:
@1ec5 I'm not seeing a proposal for fixing this line in #624.

Reply:
https://github.com/mapbox/mapbox-navigation-ios/pull/624/files#diff-eaaa217c63f0df828e942d653c6994ceL195